### PR TITLE
Make init function public to allow re-init by user

### PR DESCRIPTION
**Pull request by Manuri**
## Purpose
$Subject

Resolves https://github.com/ballerina-platform/ballerina-lang/issues/8653

### CODE REVIEW

**Comment by Manuri** [comment]
@Kishanthan As discussed offline since this should be tested at language test level, we'll not add the balo test case for this specific case.

File: tests/ballerina-unit-test/src/test/resources/test-src/connectors/h2/h2_actions_test.bal [hunk 1]
```diff
@@ -255,6 +259,39 @@ function testInitWithInvalidDbOptions() returns (int[]) {
...
+        path: "./target/H2Client/",
+        name: "TestDBH2",
+        username: "SA",
+        password: "",
+        poolOptions: { maximumPoolSize: 1 }
+    };
+
+    testDB.stop();
+
+    testDB.init(config);
```
Comment (Kishanthan) [inline tests/ballerina-unit-test/src/test/resources/test-src/connectors/h2/h2_actions_test.bal]: What if we don't call stop but directly call init again?

Comment (Manuri) [reply tests/ballerina-unit-test/src/test/resources/test-src/connectors/h2/h2_actions_test.bal]: Yes, we need to decide how to handle that scenario. I have created an issue to track this.
https://github.com/ballerina-platform/ballerina-lang/issues/10084

Comment (Kishanthan) [reply tests/ballerina-unit-test/src/test/resources/test-src/connectors/h2/h2_actions_test.bal]: Also, lets try to change this test case a bit. Lets init with different config values (for example - different db name) and then run the test case. This way we make sure that a new config is used with init.

File: tests/ballerina-unit-test/src/test/resources/test-src/connectors/sql/sql_actions_test.bal [hunk 2]
```diff
@@ -1555,3 +1555,34 @@ function testCloseConnectionPool(string jdbcUrl, string userName, string passwor
...
+    jdbc:ClientEndpointConfiguration config = {
+        url: jdbcUrl,
+        username: userName,
+        password: password,
+        poolOptions: { maximumPoolSize: 1 }
+    };
+
+    testDB.stop();
+
+    testDB.init(config);
```
Comment (Kishanthan) [inline tests/ballerina-unit-test/src/test/resources/test-src/connectors/sql/sql_actions_test.bal]: Same here. Lets try to change this test case a bit. Lets init with different config values (for example - different jdbcUrl , db name) and then run the test case. This way we make sure that a new config is used with init.

**Pull request merged**